.travis.yml

```diff
@@ -24,39 +36,59 @@ addons:
 
 matrix:
   include:
-    - compiler: gcc
+    - compiler: gcc-7
       sudo: true
       before_install:
         - sudo apt-get remove -qq libssl-dev
-      env: DTLS="OFF" ZMQ="OFF"
+      env:
+        - DTLS="OFF" ZMQ="OFF"
+        - MATRIX_EVAL="CC=gcc-7 CXX=g++-7"
       addons:
         apt:
+          sources: *toolchain_gcc
           packages:
             - *required_packages
-    - compiler: clang
+            - g++-7
+
+    - compiler: clang-5.0
       sudo: true
       before_install:
         - sudo apt-get remove -qq libssl-dev
-      env: DTLS="OFF" ZMQ="OFF"
+      env:
+        - DTLS="OFF" ZMQ="OFF"
+        - MATRIX_EVAL="CC=clang-5.0 && CXX=clang++-5.0"
       addons:
         apt:
+          sources: *toolchain_clang
           packages:
             - *required_packages
-    - compiler: gcc
-      env: DTLS="ON" ZMQ="ON"
+            - clang-5.0
+
+    - compiler: gcc-7
+      env:
+        - DTLS="ON" ZMQ="ON"
+        - MATRIX_EVAL="CC=gcc-7 CXX=g++-7"
       addons:
         apt:
-          sources: *zeromq_source
+          sources: *zeromq_source_and_toolchain_gcc
```

Comment:
instead of using
``` yaml
sources: *zeromq_source_and_toolchain_gcc
```
does this work instead:
``` yaml
sources:
  - *zeromq_source
  - *toolchain_gcc
```
I think something like that might be a little neater.

Reply:
We tried that but it failed since yaml them has lists of lists in sources. See also https://travis-ci.org/tumi8/vermont/jobs/327648923/config